examples/trajectory_follower_test/main.cpp

```diff
@@ -0,0 +1,163 @@
+// Copyright (C) 2025 COGIP Robotics association <cogip35@gmail.com>
+// This file is subject to the terms and conditions of the GNU Lesser
+// General Public License v2.1. See the file LICENSE in the top level
+// directory for more details.
+
+/**
+ * @brief Test application for TrajectoryFollowerController
+ *
+ * This example demonstrates a trajectory follower with feedforward + feedback control.
+ * It simulates a robot moving from 0 to 1000mm with:
+ * - Feedforward: pre-computed trapezoidal velocity profile
+ * - Feedback: PID correction for position tracking errors
+ */
+
+#include <cstdio>
+#include <iostream>
+
+#include "motion_control_common/ControllersIO.hpp"
+#include "motion_control_common/TrapezoidalProfile.hpp"
+
+using cogip::motion_control::ControllersIO;
+using cogip::motion_control::TrapezoidalProfile;
+
+namespace {
+
+// Simple PID class for testing (no protobuf dependencies)
+class SimplePID
+{
+  public:
+    SimplePID(double kp, double ki, double kd)
+        : kp_(kp), ki_(ki), kd_(kd), integral_(0.0), previous_error_(0.0)
+    {
+    }
+
+    double compute(double error)
+    {
+        integral_ += error;
+        double derivative = error - previous_error_;
+        previous_error_ = error;
+        return kp_ * error + ki_ * integral_ + kd_ * derivative;
+    }
+
+  private:
+    double kp_, ki_, kd_;
+    double integral_;
+    double previous_error_;
+};
+
+// Simple robot simulator
+class RobotSimulator
+{
+  public:
+    RobotSimulator() : position_(0.0), velocity_(0.0) {}
+
+    void update(double velocity_command)
+    {
+        // Simple integration: position += velocity * dt
+        // (dt = 1 period in our case)
+        velocity_ = velocity_command;
+        position_ += velocity_;
+    }
+
+    double position() const
+    {
+        return position_;
+    }
+    double velocity() const
+    {
+        return velocity_;
+    }
+
+  private:
+    double position_;
+    double velocity_;
+};
+
+} // namespace
+
+int main(void)
+{
+    std::cout << "===========================================\n";
+    std::cout << "   TrajectoryFollowerController Test\n";
+    std::cout << "===========================================\n\n";
+
+    // Create PID for position error correction
+    SimplePID pid(0.1, 0.0, 0.0); // Simple P controller
+
+    // Create trapezoidal profile
+    TrapezoidalProfile profile;
+
+    // Create robot simulator
+    RobotSimulator robot;
+
+    // Generate profile for 1000mm movement
+    double target_distance = 1000.0;
```

Comment:
Can you also add a loop with target_distance <= 0 in order to always run test in both direction ?